.github/workflows/main.yml

```diff
@@ -65,11 +65,14 @@ jobs:
         with:
           version: ${{ needs.determineVersionNumber.outputs.version }}
 
-      - name: Upload NuGetForUnity.dll
+      - name: Upload NuGetForUnity dlls
         uses: actions/upload-artifact@v3
         with:
           name: NuGetForUnity.dll
-          path: ./src/NuGetForUnity.CreateDll/bin/Release/NugetForUnity.dll
+          path: |
```

Comment:
@JoC0de Can you help me figure out why the build fails? This seems like the only change that might cause the builds to fail as they do but I don't understand why. I wrote this according to documentation for upload-artifactory action. Or maybe I am missing something in the build logs and the issue is elsewhere?

Reply:
Ok, I figured out you added a check that only allows you to change these files, but I added myself there now as well and it is still failing?

Reply:
@igor84 The GitHub Action is triggered by a `pull_request_target` event so it works on the code base of the target of the PR in this case the `master` branch. A normal `pull_request` can't be used to checkout forks as it would allow everyone to change the code that runs on the CI. I explicitly checkout the code from the merged PR but protect the CI files from being modified to mitigate the risk of anyone being able to change it.
With the current CI only I can edit `action` files as the are also fetched from the merged PR.
The `.github/workflows/main.yml` file currently is always fetched from the `master` branch so changes are not reflected until merged.
I think the only way to run the GitHub Action with your changes is if you rename your branch to match the pattern (`feature/*`):
https://github.com/GlitchEnzo/NuGetForUnity/blob/91dcd516c9b441d7c68c07158c9f12be13edbaea/.github/workflows/main.yml#L3-L6
Then the CI does a single checkout of the code from the state of the `feature/*` branch and directly executes it.

Reply:
I created a new `feature/plugin-support` branch and made a pull request from it but it still fails?

Reply:
Maybe the easiest solution would be to commit the change with my username in action.yml directly to master?

Reply:
No it works. Now every action is triggered twice:
<img width="409" alt="image" src="https://github.com/GlitchEnzo/NuGetForUnity/assets/53140583/c42836d2-5e64-4cc3-a370-91da1456e9de">
And the `push` is from your branch. And the are working....

Reply:
Ah, ok. Then I will delete this pull request and its branch. Thank you.

Reply:
Even if I commit it I can't change content of the `main.yml` and get it "executet" by a `pull_request_target` event.